back/src/main/java/com/woowacourse/teatime/teatime/repository/ScheduleRepository.java

```diff
@@ -4,13 +4,23 @@
 import java.time.LocalDateTime;
 import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 
 public interface ScheduleRepository extends JpaRepository<Schedule, Long> {
-    List<Schedule> findByCoachIdAndLocalDateTimeBetweenOrderByLocalDateTime(Long coachId, LocalDateTime start,
-                                                                            LocalDateTime end);
 
-    void deleteAllByCoachIdAndLocalDateTimeBetween(Long coachId, LocalDateTime start, LocalDateTime end);
+    @Query("SELECT s FROM Schedule AS s "
+            + "INNER JOIN s.coach AS c "
+            + "ON c.id = :coachId "
+            + "WHERE s.localDateTime >= :start "
+            + "AND s.localDateTime < :end")
+    List<Schedule> findByCoachIdBetween(Long coachId, LocalDateTime start, LocalDateTime end);
 
-    void deleteAllByCoachIdAndLocalDateTimeBetweenAndIsPossibleNot(Long coachId, LocalDateTime start, LocalDateTime end,
-                                                                   boolean isPossible);
+    @Modifying
+    @Query("DELETE FROM Schedule AS s "
+            + "WHERE s.coach.id = :coachId "
+            + "AND s.isPossible = true "
+            + "AND s.localDateTime >= :start "
+            + "AND s.localDateTime < :end")
+    void deleteAllReservableByCoachIdBetween(Long coachId, LocalDateTime start, LocalDateTime end);
```

Comment:
between이면 <= end로 해야하는 거 아닐까요?

Reply:
[Mysql 8.0 공식문서](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_between)를 읽어보니 마루 말이 맞네요!

> If expr is greater than or equal to min and expr is less than or equal to max, [BETWEEN](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_between) returns 1, otherwise it returns 0.  
This is equivalent to the expression (min <= expr AND expr <= max)

비즈니스 로직에는 마지막 end 부등호는 크게 영향이 없어서 네이밍은 between일때는 부등호를 (>= start, < end)로 통일할게요!